### ISSUE TITLE
Use action/cache in GitHub Actions docs

### ISSUE THREAD

**Issue by simonw** [astral-sh/uv#6292]
These docs currently suggest: https://docs.astral.sh/uv/guides/integration/github/
```yaml
    steps:
      - uses: actions/checkout@v4

      - name: Set up uv
        # Install a specific uv version using the installer
        run: curl -LsSf https://astral.sh/uv/0.3.0/install.sh | sh
```
I’m not a fan of that pattern without caching - it feels wasteful to download from outside the GitHub network on every run, plus an `astral.sh` outage would break my builds.

It would be nice if this example used `action/cache` to avoid duplicate downloads where possible.

**Comment by simonw** [comment]
Alternatively this recipe could use this URL instead:

https://github.com/astral-sh/uv/releases/download/0.3.0/uv-installer.sh

Which would at least keep all network traffic within GitHub.

**Comment by zanieb** [comment]
I think we'll probably provide a `setup-uv` action in the future that manages installation, presumably that could handle caching. I'm pretty hesitant to complicate the integration guide here, but am willing to consider using `action/cache` in a subsequent example if you want to share how you would do it.

**Comment by simonw** [comment]
`setup-uv` action would be fantastic - or even getting `uv` into the default runner, but that's GitHub's call.

I messed around with caching a bit and [got this to work](https://github.com/simonw/playing-with-actions-single/blob/09040ddd30c0a85d03d4d4c472def0a32a23fef6/.github/workflows/uv.yaml):
```yaml
name: uv example

on:
  workflow_dispatch:

jobs:
  uv-example:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v4
    - id: cache-cargo-bin
      uses: actions/cache@v4
      with:
        path: ~/.cargo/bin
        key: ${{ runner.os }}-cargo-bin
    - id: cache-uv-python
      uses: actions/cache@v4
      with:
        path: ~/.local/share/uv
        key: ${{ runner.os }}-uv-python
    - name: Install uv
      if: steps.cache-cargo-bin.outputs.cache-hit != 'true'
      run: |
        curl -LsSf https://astral.sh/uv/0.3.0/install.sh | sh
        cd $HOME
        find .cargo
    - name: Run uv
      run: |
        python -c 'import sys; print("Before:", sys.version)'
        uv python install
        cd $HOME
        uv run python -c 'import sys; print(sys.version)'
```
The second time I ran that the output looked like this, showing the cache working:

![CleanShot 2024-08-20 at 20 22 53@2x](https://github.com/user-attachments/assets/bc52e100-3e8a-4799-bfa7-1c0f7e66aafc)

There may be ways to make this shorter and tighter.

**Comment by skshetry** [comment]
I have been using https://github.com/hynek/setup-cached-uv, and it's been working well for me. :)

**Comment by zanieb** [comment]
Note that `setup-cached-uv` is different in that it persists the uv cache to GitHub.

**Comment by simonw** [comment]
An official Astral version of that reusable Action would be really neat. My interest is primarily in avoiding hitting external network resources for repeat runs of my CI builds.

**Comment by danielhollas** [comment]
> I’m not a fan of that pattern without caching - it feels wasteful to download from outside the GitHub network on every run, plus an astral.sh outage would break my builds.

Given Github Actions track record I'd honestly be more wary of github cache outage in this case. :-D 

But more seriously, I am curious if measured any speed difference between these approaches?

**Comment by zanieb** [comment]
uv seems to download and install in [~1 second](https://github.com/astral-sh/packse/actions/runs/10316016498/job/28557350216)

**Comment by zanieb** [comment]
Resolved by #7056
